### PR TITLE
[symbology] Fix SVG marker anchor calculation and improve bounds

### PR DESCRIPTION
**Pull request by nirvn**
## Description

This PR fixes a long-standing issue with SVG markers whereas its anchor calculation was wrongly assuming the SVG marker to be square in size. It resulted in wrong placement with setting the vertical anchor to top / bottom.

### CODE REVIEW

File: src/core/symbology/qgsmarkersymbollayer.cpp [hunk 1]
```diff
@@ -2458,18 +2459,24 @@ bool QgsSvgMarkerSymbolLayer::writeDxf( QgsDxfExport &e, double mmMapUnitScaleFa
 QRectF QgsSvgMarkerSymbolLayer::bounds( QPointF point, QgsSymbolRenderContext &context )
 {
   bool hasDataDefinedSize = false;
-  double scaledSize = calculateSize( context, hasDataDefinedSize );
-  scaledSize = context.renderContext().convertToPainterUnits( scaledSize, mSizeUnit, mSizeMapUnitScale );
+  double scaledWidth = calculateSize( context, hasDataDefinedSize );
```
Comment (nyalldawson) [inline src/core/symbology/qgsmarkersymbollayer.cpp:2462]: I don't think these changes correctly account for data defined symbol names...?

Comment (nirvn) [reply src/core/symbology/qgsmarkersymbollayer.cpp:2462]: Hmm, you are right. I guess there's no way around calling the svg cache to get height for data defined names/paths. 

This means to get proper anchoring, we'll have to do the same in renderPoint too. You're good with that?

**Comment by nirvn** [comment]
@nyalldawson , code updated.

File: src/core/symbology/qgsmarkersymbollayer.cpp [hunk 2]
```diff
@@ -2487,29 +2494,37 @@ QRectF QgsSvgMarkerSymbolLayer::bounds( QPointF point, QgsSymbolRenderContext &c
   }
   strokeWidth = context.renderContext().convertToPainterUnits( strokeWidth, mStrokeWidthUnit, mStrokeWidthMapUnitScale );
 
-  //need to get colors to take advantage of cached SVGs
```
Comment (nyalldawson) [inline src/core/symbology/qgsmarkersymbollayer.cpp:2490]: can you leave this comment? I found it useful to know why the colors are being decoded here

**Review by nyalldawson — APPROVED** [review]

File: src/core/symbology/qgsmarkersymbollayer.cpp (hunk 2 above)
Comment (nirvn) [reply src/core/symbology/qgsmarkersymbollayer.cpp:2490]: @nyalldawson , will re-add -- thanks for the review, the SVG marker code logic can be a bit of a PITA.

**Pull request merged**